custom_components/rivian/coordinator.py

```diff
@@ -103,18 +103,25 @@ class ChargingCoordinator(RivianDataUpdateCoordinator[dict[str, Any]]):
     """Charging data update coordinator for Rivian."""
 
     key = "getLiveSessionData"
+    _unplugged_interval = 15 * 60 # 15 minutes
+    _plugged_interval = 30 # 30 seconds
+    _update_interval = _unplugged_interval  # 15 minutes
 
-    def __init__(self, hass: HomeAssistant, client: Rivian, vin: str) -> None:
+    def __init__(self, hass: HomeAssistant, client: Rivian, vehicle_id: str) -> None:
         """Initialize the coordinator."""
         super().__init__(hass=hass, client=client)
-        self.vin = vin
+        self.vehicle_id = vehicle_id
 
     async def _fetch_data(self) -> ClientResponse:
         """Fetch the data."""
         return await self.api.get_live_charging_session(
-            vin=self.vin, properties=CHARGING_API_FIELDS
+            vin=self.vehicle_id, properties=CHARGING_API_FIELDS
         )
 
+    def adjust_update_interval(self, is_plugged_in: bool) -> None:
+        """Adjust update interval based on plugged in status."""
+        self._set_update_interval(self._plugged_interval if is_plugged_in else self._unplugged_interval)
```

Comment:
This isn't called from anywhere, so you'll need to backport the section that calls this as well.

Reply:
@natekspencer as you have been more intimately in the coordinator logic, check my understanding of the latest push.

I pulled reference to the `ChargingCoordinator` within `VehicleCoordinator` so updates to the wake and plugged in state can be pushed into the `ChargingCoordinator.adjust_update_interval(...)` . In doing so I changed a bit of the `__init__.py` logic for associating these together.